src/app.rs

```diff
@@ -1,255 +1,268 @@
-        let deletion = HistoryOperation::DeleteElement {
-            text: deleted_node.text,
-            tree_path,
-            children: deleted_node.children,
-        };
+    pub fn get_back_to_parent(&mut self) -> Option<()> {
+        let current_parent_task_id = self.opened_task?;
+        let current_parent_task_entity = self.repository.get_task(current_parent_task_id)?;
+        let next_parent_task_id = current_parent_task_entity.parent_id;
 
-        self.history.push_operation(deletion);
+        self.opened_task = next_parent_task_id;
+        self.scroll_to_top();
+        Some(())
     }
 
-    fn delete_at(&mut self, tree_path: &[usize]) -> ElementTree {
-        if tree_path.len() == 1 {
-            self.trees.remove(tree_path[0])
-        } else {
-            self.trees[tree_path[0]].remove_node(&tree_path[1..])
-        }
+    pub fn init_insert_mode_to_edit_a_task_title(&mut self) -> Option<()> {
+        let selected_task = self.get_selected_task()?;
+
+        let task_id = selected_task.id;
+        let title_to_edit = selected_task.title.clone();
+
+        self.text_area = TextArea::from([title_to_edit]);
+        self.text_area.move_cursor(tui_textarea::CursorMove::End);
+        self.state = AppState::INSERT(task_id);
+        Some(())
     }
 
-    fn insert_at(&mut self, tree_path: &[usize], new_leaf: ElementTree) {
-        if tree_path.len() == 1 {
-            self.trees.insert(tree_path[0], new_leaf);
-        } else {
-            self.trees[tree_path[0]].insert_node(&tree_path[1..], new_leaf);
+    pub fn cancel_insert_mode(&mut self) {
+        self.state = AppState::NORMAL;
+    }
+
+    pub fn close_insert_mode_updating_task_title(&mut self) {
+        if let AppState::INSERT(task_id) = self.state {
+            let content = self.text_area.lines().join("\n");
+            if !content.is_empty() {
+                self.repository.update_task_title(task_id, content);
+            }
+
+            self.state = AppState::NORMAL;
         }
     }
+}
 
-    pub fn scroll_to_top(&mut self) {
-        self.update_selection(0);
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_new_app() {
+        let repository = AppTreeRepository::default();
+        let app = App::new(repository);
+
+        assert_eq!(app.selection_index, 0);
+        assert!(matches!(app.state, AppState::NORMAL));
+        assert!(app.opened_task.is_none());
     }
 
-    pub fn scroll_to_bottom(&mut self) {
-        self.update_selection(self.trees.len() - 1);
+    #[test]
+    fn test_add_new_task() {
+        let repository = AppTreeRepository::default();
+        let mut app = App::new(repository);
+
+        app.add_new_task();
+        let tasks = app.find_tasks_to_display();
+        assert_eq!(app.opened_task, None);
+        assert_eq!(tasks.len(), 1);
+
+        // This sleep is to prevent ID collision
+        // Current the ID is generated by the system time (unix timestamp)
```

Comment:
hey can you use incremental ids so we don't bump into this

every time we need an id the generator +=1 and returns it

compared to Uuid::new_v4(), when loading stuff from storage, we need to pick the largest that was read, so that's a con, but as a pro it helps debugability, and you don't have huge ids